packages/front-end/services/SQLInputField.tsx

```diff
@@ -0,0 +1,223 @@
+import { UserIdType } from "back-end/types/datasource";
+import React, { ReactElement, useEffect, useState } from "react";
+import { UseFormReturn } from "react-hook-form";
+import { FaPlay } from "react-icons/fa";
+import CodeTextArea from "../components/Forms/CodeTextArea";
+import DisplayTestQueryResults from "../components/Settings/DisplayTestQueryResults";
+import Code from "../components/SyntaxHighlighting/Code";
+import Tooltip from "../components/Tooltip/Tooltip";
+import { useAuth } from "./auth";
+import { validateSQL } from "./datasources";
+import { TestQueryRow } from "../../back-end/src/types/Integration";
+
+type TestQueryResults = {
+  duration?: string;
+  error?: string;
+  results?: TestQueryRow[];
+  sql?: string;
+};
+
+type Props = {
+  userEnteredQuery: string;
+  datasourceId: string;
+  // eslint-disable-next-line
+  form: UseFormReturn<any>;
```

Comment:
I've still not grasped what value should go here after reading the documentation. Any guidance is appreciated.

Reply:
I'm not a TS expert, but I was thinking it might be a union of your component's mandatory fields and a passed in generic from each component which embeds <SQLInputField>. I tried annotating the component's props with the generic <T> and passing it from the embedder via something like `<SQLInputField<DimensionFormInterface>` and receiving it via:
```
type Props<T> = {
...
  form: UseFormReturn<FormInterface & T>;
...
};

export default function SQLInputField<T>({
...
}: Props<T>) {
```
, but was getting some errors doing so. Might work with a bit more effort, or might just be a rabbit hole...